src/hooks/useDailyStatistics/useDailyStatistics.test.tsx

```diff
@@ -0,0 +1,247 @@
+    order: 1,
+    identifiers: [
+      {
+        ...defaultIdentifier,
+        label: "first"
+      }
+    ],
+    quantity: {
+      period: -1,
+      periodType: "ROLLING",
+      periodExpression: 365,
+      limit: 1,
+      default: 1
+    },
+    type: "REDEEM"
+  },
+  {
+    category: "specs-lost",
+    categoryType: "APPEAL",
+    name: "Lost Specs",
+    order: 1,
+    alert: {
+      threshold: 2,
+      label: "*chargeable"
+    },
+    quantity: {
+      period: -1,
+      periodType: "ROLLING",
+      periodExpression: 365,
+      limit: 9999,
+      default: 1
+    },
+    identifiers: [
+      {
+        ...defaultIdentifier,
+        label: "first"
+      }
+    ],
+    type: "REDEEM"
+  }
+];
+
+describe("useDailyStatistics", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    mockGetDailyStatistics.mockReset();
+  });
+
+  it("should set error if error is thrown while getting daily stats", async () => {
+    expect.assertions(1);
+
+    mockGetDailyStatistics.mockRejectedValue(
+      new Error("Error getting daily statistics")
+    );
+
+    const { result, waitForNextUpdate } = renderHook(
+      () => useDailyStatistics(key, endpoint, operatorToken, Date.now()),
+      { wrapper }
+    );
+
+    await waitForNextUpdate();
+
+    expect(result.current.error?.message).toBe(
+      "Error getting daily statistics"
+    );
+  });
+
+  it("should set last transaction time, total count, transaction history after getting daily stats", async () => {
+    expect.assertions(3);
+
+    mockGetDailyStatistics.mockResolvedValue({
+      pastTransactions: [
+        {
+          category: "vouchers",
+          quantity: 9999999,
+          transactionTime: new Date(12000000000)
+        }
+      ]
+    });
+
+    const { result, waitForNextUpdate } = renderHook(
+      () => useDailyStatistics(key, endpoint, operatorToken, 12000000000),
+      { wrapper }
+    );
+
+    await waitForNextUpdate();
+
+    expect(result.current.lastTransactionTime).toStrictEqual(
+      new Date(12000000000)
+    );
+    expect(result.current.totalCount).toBe(9999999);
+    expect(result.current.transactionHistory).toStrictEqual([
+      {
+        category: "vouchers",
+        name: "vouchers",
+        quantityText: "9,999,999 qty"
+      }
+    ]);
+  });
+
+  it("last transaction time should be set as null if transaction history is empty", async () => {
+    expect.assertions(2);
+
+    mockGetDailyStatistics.mockResolvedValue({
+      pastTransactions: []
+    });
+
+    const { result, waitForNextUpdate } = renderHook(
+      () => useDailyStatistics(key, endpoint, operatorToken, 12000000000),
+      { wrapper }
+    );
+
+    await waitForNextUpdate();
+
+    expect(result.current.lastTransactionTime).toBeNull();
+    expect(result.current.transactionHistory).toStrictEqual([]);
+  });
+
+  it("should not call fetchDailyStatistics again if timestamp remains the same, instead should be unable to render, and timeout", async () => {
```

Comment:
Hmm why would this cause inability to render and timeout ah? In what cases might user encounter this, and what would user see?